types/lodash/common/object.d.ts

```diff
@@ -1021,41 +1021,115 @@ declare module "../index" {
         functionsIn(): CollectionChain<string>;
     }
 
-    type GetIndexedField<T, K> = K extends keyof T
-        ? T[K]
-        : K extends `${number}`
-            ? 'length' extends keyof T
-                ? number extends T['length']
-                    ? number extends keyof T
-                        ? T[number]
-                        : undefined
-                    : undefined
-                : undefined
-            : undefined;
+    type GetElementTypeOfFixedSizeArrayLikeByIndex<T extends unknown[], I extends number> =
+        number extends I ? T[I] | undefined : T[I];
```

Comment:
While `T[I] | undefined` is technically correct, it is a breaking change and inconsistent with the rest of TypeScript, which omits `| undefined` from the type of array indexes. So we should remove `|undefined` here.